vars/tagRepo.groovy

```diff
@@ -7,35 +7,30 @@ def call(String gitRepoUrl, String checkoutGitBranch, String checkoutDir, String
         targetDir = checkoutDir
     }
     dir(checkoutDir) {
-        sshagent(['jenkinsgithub']) {
-            //How do we get this from the global config? Is sshagent the only way to do this currently or is there a git pipeline step?
-            sh "git config user.name \"Jenkins\""
-            sh "git config user.email \"jenkins@wendy.feedhenry.net\""
 
-            body()
+        body()
 
-            def existingTagCommitHash = sh(returnStdout: true, script: "git ls-remote origin refs/tags/${tagName} | cut -f 1").trim()
-            def tagCommitHash = sh(returnStdout: true, script: "git rev-parse ${tagGitRef}").trim()
+        def existingTagCommitHash = sh(returnStdout: true, script: "git ls-remote origin refs/tags/${tagName} | cut -f 1").trim()
+        def tagCommitHash = sh(returnStdout: true, script: "git rev-parse ${tagGitRef}").trim()
 
-            if (existingTagCommitHash) {
-                if (existingTagCommitHash == tagCommitHash) {
-                    print "Tag ${tagName} already exists and tagCommitHash(${tagCommitHash}) is the same, can continue!"
-                } else {
-                    print "Tag ${tagName} already exists but tagCommitHash(${tagCommitHash}) is not the same, can't continue!"
-                    sh('exit 1')
-                }
+        if (existingTagCommitHash) {
+            if (existingTagCommitHash == tagCommitHash) {
+                print "Tag ${tagName} already exists and tagCommitHash(${tagCommitHash}) is the same, can continue!"
             } else {
-                print "Creating new tag ${tagName} at ${tagCommitHash}"
-                sh "git tag ${tagName} ${tagCommitHash}"
+                print "Tag ${tagName} already exists but tagCommitHash(${tagCommitHash}) is not the same, can't continue!"
+                sh('exit 1')
             }
+        } else {
+            print "Creating new tag ${tagName} at ${tagCommitHash}"
+            sh "git tag ${tagName} ${tagCommitHash}"
+        }
 
-            if (params.dryRun) {
-                print "Would push ${checkoutGitBranch} to ${gitRepoUrl}"
-                print "Would push ${tagName} to ${gitRepoUrl}"
-            } else {
-                sh "git push origin HEAD:${checkoutGitBranch}"
-                sh "git push origin ${tagName}"
-            }
+        if (params.dryRun) {
+            print "Would push ${checkoutGitBranch} to ${gitRepoUrl}"
+            print "Would push ${tagName} to ${gitRepoUrl}"
+        } else {
+            sh "git push origin HEAD:${checkoutGitBranch}"
+            sh "git push origin ${tagName}"
         }
     }
 }
```

Comment:
Remove sshagent and git config setup. This can be really slow if you are calling this method in a parallel step. Configure the node first and call this:

```
node('nodejs') {
        step([$class: 'WsCleanup'])
        ansiColor('xterm') {
            sshagent(['jenkinsgithub']) {
                sh "git config --global user.name \"Jenkins\""
                sh "git config --global user.email \"jenkins@wendy.feedhenry.net\""
                tagRepo(gitUrl, gitBranch, 'blah', gitTag, gitTag2)
        }
}
```